pages/login.page.js

```diff
@@ -0,0 +1,29 @@
+const PAGE_TIMEOUT = 10000;
+
+class LoginPage {
+    
+    get usernameInput () { return $('#username'); }
+    get passwordInput () { return $('#password'); }
+    get signInButton () { return $('#submit'); }
+    get rememberMe () { return $('#remember-me'); }
+    get alert () {return $('[class= "badge badge-pill badge-danger"]'); }
+
+
+    async logIn(username, password) {
+        await this.usernameInput.waitForClickable({ timeout: PAGE_TIMEOUT });
+        await this.usernameInput.setValue(username);
+        await this.passwordInput.waitForClickable({ timeout: PAGE_TIMEOUT });
+        await this.passwordInput.setValue(password);
```

Comment:
There is duplication of the pattern of waiting for an element to be clickable and setting their value here. Consider encapsulating that logic into its own method: Something like `completeField(element, text)` that waits for an element to be clickable and then sends it the string you pass it by parameter. That way you reduce overall lines for `logIn()` method and make it more readable (as you make it work with constructs of higher-level of abstraction).

Also, this `logIn()` method is not actually logging in, it's just filling the form fields (not pressing enter nor clicking the submit button). Make sure you provide implementations for that as well (something like `logInWithEnter()` and `logInWithSubmitButton()`)